client/src/components/ProfileForm.tsx

```diff
@@ -0,0 +1,173 @@
+import { useState } from "react";
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  Input,
+  Textarea,
+  useToast,
+} from "@chakra-ui/react";
+import { Profile } from "types/Profile";
+import { useSelector } from "react-redux";
+import { userSelector } from "redux/user";
+
+interface UserData {
+  displayName: string;
+  email: string;
+}
+
+export const ProfileForm = (props: {
+  isOpen: boolean;
+  onClose: () => void;
+  isEditing: boolean;
+  setIsEditing: React.Dispatch<React.SetStateAction<boolean>>;
+  userData: UserData | null | undefined;
+  profileData: Profile;
+}) => {
+  const API_URL = `${import.meta.env.VITE_API_SERVER_URL}/api/v1`;
+  const AuthToken = localStorage.getItem("BEARER_TOKEN") || "";
+  const [formState, setFormState] = useState({
+    displayName: props.userData?.displayName || "",
+    email: props.userData?.email || "",
+    avatarUrl: props.profileData.avatarUrl || "",
+    bio: props.profileData.bio || "",
+  });
+  const toast = useToast();
+  const { user } = useSelector(userSelector);
+
+  const handleSubmit = async () => {
+    const emailRegex =
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    if (!emailRegex.test(formState.email)) {
+      throw new Error("Invalid email address");
+    }
+    if (!formState.avatarUrl) {
+      throw new Error("Invalid email address");
+    }
+    try {
+      // Update user display name and email.
+      const userResponse = await fetch(`${API_URL}/users/${user?.id}`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${AuthToken}`,
+        },
+        body: JSON.stringify({ formState }),
+      });
+
+      if (!userResponse.ok) {
+        throw new Error("Error updating user data");
+      }
+
+      // Update user profile data.
+      const profileResponse = await fetch(`${API_URL}/users/profile/${user?.id}`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${AuthToken}`,
+        },
+        body: JSON.stringify({ formState }),
+      });
```

Comment:
If you are sending 2 similar information to 2 different endpoints, instead just do 1 endpoint and process all of them.

You can set up your user update endpoint to accept the avatarUrl and bio field and update them in the bio as well, see https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-writes